packages/react/src/contexts/provider.ts

```diff
@@ -0,0 +1,37 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import * as React from 'react'
+import { AbstractConfigContext, AbstractConfigProps } from './config'
+
+const client = new QueryClient({
+  defaultOptions: {
+    queries: {
+      cacheTime: 1_000 * 60 * 60 * 24, // 24 hours
+      networkMode: 'offlineFirst',
+      refetchOnWindowFocus: false,
+      retry: 0,
+    },
+    mutations: {
+      networkMode: 'offlineFirst',
+    },
+  },
+})
+
+export type AbstractProviderProps = AbstractConfigProps
```

Comment:
We should probably allow the consumer to provide query client options as well - this is what graz does https://github.com/graz-sh/graz/blob/2dd73ddae3313e76be5f79215a30c0916c9c1fcd/packages/graz/src/provider/index.tsx#L2

Reply:
or a client should instead init the query client on his side and inject it into the provider, we could build a fancy `createConfig` as wagmi did.